---
title: "Estimating cell-type composition using Gene Expression data "
author: "S Karkar, M Richard, Y Blum"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
keep_md: true
vignette: >
  %\VignetteIndexEntry{Deconvolution}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Based on Gene Expression values, the vignette shows how to estimate cell-type composition and cell type-specific expression profiles.

## Data

We provide access to a matrix of Gene Expression values `D` from a pseudobulk, that is, a simulation of complex tissue comprised of $k=9$ cell types.
Details regarding the genertaion  

```{r datas}
D = readRDS(url(""))
dim(D)
head(D)[1:5, 1:5]

```


## Step 1: Normalising data

In addition to cell-type composition, Gene Expression values can vary because of other technical variables, such as batch effects and, for RNA-Seq, read depth. and We use the function `run_norm` to account for this effect. By default, the function assumes that the normalizing method is 'RPM', In addition we provide access to 3 other methods "TPM", "DESeq2" (and its in-house rewriting : "MR" that do do required the `DESeq2` package), and "edgeR". Note that "TPM" requires the gene length (in bp) in form of a vector of appropriate length.   

  
```{r run_norm}
 
```

## Step 2: transforming data 

We provide the function `run_trans`, which performs a transformation of the data. By default, the transformation method is "linear", which simply return the input data. Additionally, we provide 2 transformations : "log2" and "pseudolog". "log2" transformation apply the mathematical function $f(x)=log2(x+1)$ where $x$ is the input data, and "pseudolog" corresponds to the function $f(x)=asinh(x)$  

 
```{r run_trans}
 
```


## Step 3: feature selection 

The function `run_featsel` select vataiables (genes) with the largest Coefficient of Variation.
We propose to select 5000 genes ("cv5000") or 1000 genes ("cv1000"). By removing genes that carry little or no information, deconvolution routines run much faster.

```{r run_featsel}
 
```

## Step 4: running deconvolution methods

By default, we propose 2 methods of deconvolution : `NMF` to run `NMF::snmf`, `ICA` to run `deconica` package. Additionally, when available on the system, `CDSeq` and `PREDE` can be used. 

To select the number of principal components, we use Cattell rule that recommends to keep principal components to the left of the elbow in the eigenvalue plot. In practice Cattell rule suggests to keep the number of principal components at the observed elbow plus one.

Here we show the results obtained with `NMF`.
```{r deconv}
results_RFE = medepir::RFE(D_FS, nbcell = 5)
```



## Step 5: Enrichment analysis.

```{r enrich}

```

