---
title: "Estimating cell-type composition using Gene Expression data"
author: "S Karkar, M Richard, Y Blum"
date: "`r Sys.Date()`"
output:
  rmarkdown::html_vignette: default
  html_document:
    df_print: paged
  pdf_document: default
keep_md: yes
vignette: |
  %\VignetteIndexEntry{Estimating cell-type composition using Gene Expression data} %\VignetteEngine{knitr::rmarkdown} %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Based on Gene Expression values, the vignette shows how to estimate cell-type composition and cell type-specific expression profiles.

## Data

We provide access to a matrix of Gene Expression values `D` from a pseudo-bulk, that is, a simulation of complex tissue comprised of $k=9$ cell types.


```{r datas}
D = readRDS(url("https://figshare.com/ndownloader/files/30587814"))
dim(D)
head(D)[1:5, 1:5]

```


## Step 1: Normalizing data

In addition to cell-type composition, Gene Expression values can vary because of other technical variables, such as, for RNA-Seq, read depth. We use the function `run_norm` to account for this effect. By default, the function assumes that the normalizing method is "RPM", In addition we provide access to 3 other methods "TPM", "DESeq2" (and its in-house rewriting : "MR" that do do required the **DESeq2** package), and "edgeR". Note that "TPM" requires the gene length (in bp) in form of a vector of appropriate length.   

  
```{r run_norm}
library(gedepir)
 D_norm=run_norm(mix_matrix = D,method = "RPM")
```

## Step 2: Transforming data 

We provide the function `run_trans`, which performs a transformation of the data. By default, the transformation method is "linear", which simply return the input data. Additionally, we provide 2 transformations : "log2" and "pseudolog". "log2" transformation apply the mathematical function $f(x)=log2(x+1)$ where $x$ is the input data, and "pseudolog" corresponds to the function $f(x)=asinh(x)$  

 
```{r run_trans}
 D_trans= run_trans(mix_matrix = D_norm,method = "log2")
```


## Step 3: Feature selection 

The function `run_featsel` select vataiables (genes) with the largest Coefficient of Variation.
We propose to select 5000 genes ("cv5000") or 1000 genes ("cv1000"). By removing genes that carry little or no information, deconvolution routines run much faster.

```{r run_featsel}
 D_fsel= run_featsel(mix_matrix = D_trans, method = "cv1000")
```

## Step 4: Running deconvolution methods

By default, we propose 2 methods of deconvolution: "NMF" to run `NMF::snmf`, "ICA" from the *deconica* package. Additionally, when available on the system, "CDSeq" and "PREDE" methods can be used. 

To select the number of cell types, Cattell's rule uses PCA analysis and recommends to keep principal components to the left of the elbow in the eigenvalue plot. In practice Cattell rule suggests to keep the number of principal components at the observed elbow plus one.

Here we show the results obtained with `NMF`.
```{r deconv}
results_NMF = run_deconv(mix_matrix = D_fsel,k= 9, method = "NMF")
```



## Step 5: Enrichment analysis
`gedepir` includes a wrapper for `fgsea` to analyze  the estimated cell type profiles.

We provide several databases available for enrichment analysis available on [figShare](https://figshare.com/articles/dataset/Database_for_enrichment_analysis/16546470)

Database| FileName | url | Description 
-----|-----|-----|-----:
CellMarker| 'cellmarkerDB.RDS'| [link](https://figshare.com/ndownloader/files/30904705) | CellMarker (Zhang et al., 2019, DOI: 10.1093/nar/gky900), human restricted marker file
CEllMatch|'cellmatchDB.RDS' |[link](https://figshare.com/ndownloader/files/30904711) |  CEllMatch (Shao et al, 2020,DOI: 10.1016/j.isci.2020.100882, included in the [scCATCH tool](https://github.com/ZJUFanLab/scCATCH/tree/master/R)
NCI-60  |  'Cancer_Cell_Lines.RDS' | [link](https://figshare.com/ndownloader/files/30593271) | NCI-60 Human Tumor Cell Lines Screen 
GTEx | 'GTEx.RDS'|[link](https://figshare.com/ndownloader/files/30593274)|  Genotype-Tissue Expression project, specific expression (up and down) for 2918 tissue samples
Reactome | 'Reactome_2016.RDS' | [link](https://figshare.com/ndownloader/files/30593277) |Reactome db, free, curated and peer-reviewed pathway database.  v.2016
GO db|  'GO.DB.RDS'  | [link](https://figshare.com/ndownloader/files/30593280)| Gene Ontology database,  annotation for functions of genes
Kegg  |  'KEGG_2019_Human.RDS' | [link](https://figshare.com/ndownloader/files/30593283) |   Kyoto Encyclopedia of Genes and Genomes database, high-level annotations from molecular-level information, v.2019
  



```{r enrich}
library(fgsea)
database= readRDS(url("https://figshare.com/ndownloader/files/30593259"))
gedepir::enrich(results_NMF$T_matrix,pathways = database,ICAbased = FALSE, fdr = FALSE)
```

## Step 6: Build your pipeline
**gedepir** simplifies the definition of an end-to-end analysis pipeline with a set of base functions that are connected through the *pipes* syntax `  %>% `  used in **magrittr**,   **tidyr** or   **dplyr**  packages.
```{r pipeline}
library(magrittr)
results_ICA= D %>%
  run_norm(method = "RPM") %>%
  run_trans(method = "linear") %>%
  run_deconv(method = "ICA")
gedepir::enrich(results_ICA$T_matrix,database,ICAbased = TRUE)
```

