<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="S Karkar, M Richard, Y Blum" />

<meta name="date" content="2021-09-02" />

<title>Estimating cell-type composition using Gene Expression data</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Estimating cell-type composition using Gene Expression data</h1>
<h4 class="author">S Karkar, M Richard, Y Blum</h4>
<h4 class="date">2021-09-02</h4>



<p>Based on Gene Expression values, the vignette shows how to estimate cell-type composition and cell type-specific expression profiles.</p>
<div id="data" class="section level2">
<h2>Data</h2>
<p>We provide access to a matrix of Gene Expression values <code>D</code> from a pseudo-bulk, that is, a simulation of complex tissue comprised of <span class="math inline">\(k=9\)</span> cell types.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>D <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">url</span>(<span class="st">&quot;https://figshare.com/ndownloader/files/30587814&quot;</span>))</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(D)</span></code></pre></div>
<pre><code>## [1] 26485    50</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(D)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##                 M1           M2          M3       M4       M5
## DDX11L1   1.001393  0.004283769 0.001109519 4.001646 1.000574
## WASH7P    8.967564 11.902306950 6.890561585 9.200173 7.142057
## MIR6859-3 2.000000  1.000000000 0.000000000 0.000000 0.000000
## MIR6859-2 1.000000  1.000000000 3.000000000 0.000000 1.000000
## MIR6859-1 0.000000  0.000000000 0.000000000 0.000000 0.000000</code></pre>
</div>
<div id="step-1-normalising-data" class="section level2">
<h2>Step 1: Normalising data</h2>
<p>In addition to cell-type composition, Gene Expression values can vary because of other technical variables, such as batch effects and, for RNA-Seq, read depth. and We use the function <code>run_norm</code> to account for this effect. By default, the function assumes that the normalizing method is “RPM”, In addition we provide access to 3 other methods “TPM”, “DESeq2” (and its in-house rewriting : “MR” that do do required the <strong>DESeq2</strong> package), and “edgeR”. Note that “TPM” requires the gene length (in bp) in form of a vector of appropriate length.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gedepir)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a> D_norm<span class="ot">=</span><span class="fu">run_norm</span>(<span class="at">mix_matrix =</span> D,<span class="at">method =</span> <span class="st">&quot;RPM&quot;</span>)</span></code></pre></div>
</div>
<div id="step-2-transforming-data" class="section level2">
<h2>Step 2: transforming data</h2>
<p>We provide the function <code>run_trans</code>, which performs a transformation of the data. By default, the transformation method is “linear”, which simply return the input data. Additionally, we provide 2 transformations : “log2” and “pseudolog”. “log2” transformation apply the mathematical function <span class="math inline">\(f(x)=log2(x+1)\)</span> where <span class="math inline">\(x\)</span> is the input data, and “pseudolog” corresponds to the function <span class="math inline">\(f(x)=asinh(x)\)</span></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a> D_trans<span class="ot">=</span> <span class="fu">run_trans</span>(<span class="at">mix_matrix =</span> D_norm,<span class="at">method =</span> <span class="st">&quot;log2&quot;</span>)</span></code></pre></div>
</div>
<div id="step-3-feature-selection" class="section level2">
<h2>Step 3: feature selection</h2>
<p>The function <code>run_featsel</code> select vataiables (genes) with the largest Coefficient of Variation. We propose to select 5000 genes (“cv5000”) or 1000 genes (“cv1000”). By removing genes that carry little or no information, deconvolution routines run much faster.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a> D_fsel<span class="ot">=</span> <span class="fu">run_featsel</span>(<span class="at">mix_matrix =</span> D_trans,<span class="at">method =</span> <span class="st">&quot;cv1000&quot;</span>)</span></code></pre></div>
</div>
<div id="step-4-running-deconvolution-methods" class="section level2">
<h2>Step 4: running deconvolution methods</h2>
<p>By default, we propose 2 methods of deconvolution : “NMF” to run <code>NMF::snmf</code>, “ICA” from the <em>deconica</em> package. Additionally, when available on the system, “CDSeq” and “PREDE” methods can be used.</p>
<p>To select the number of cell types, Cattell rule uses PCA analysis and recommends to keep principal components to the left of the elbow in the eigenvalue plot. In practice Cattell rule suggests to keep the number of principal components at the observed elbow plus one.</p>
<p>Here we show the results obtained with <code>NMF</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>results_NMF <span class="ot">=</span> <span class="fu">run_deconv</span>(<span class="at">mix_matrix =</span> D_fsel,<span class="at">k=</span> <span class="dv">9</span>, <span class="at">method =</span> <span class="st">&quot;NMF&quot;</span>)</span></code></pre></div>
</div>
<div id="step-5-enrichment-analysis." class="section level2">
<h2>Step 5: Enrichment analysis.</h2>
<p><code>gedepir</code> includes a wrapper for <code>fgsea</code> to analyze the estimated cell type profiles.</p>
<p>We provide several databases available for enrichment analysis available at : [[<a href="https://figshare.com/articles/dataset/Database_for_enrichment_analysis/16546470" class="uri">https://figshare.com/articles/dataset/Database_for_enrichment_analysis/16546470</a>]]</p>
<ul>
<li>‘Tissue.markers.RDS’</li>
<li>‘Cancer.markers.RDS’</li>
<li>‘Tissue.CellTypes.markers.RDS’</li>
<li>‘Cancer_CellMatch.RDS’</li>
<li>‘Cancer_Cell_Lines.RDS’</li>
<li>‘GTEx.RDS’</li>
<li>‘Reactome_2016.RDS’</li>
<li>‘GO.DB.RDS’</li>
<li>‘KEGG_2019_Human.RDS’</li>
<li>‘markerscelltypes.RDS’</li>
<li>‘CellMatchCurated.RDS’</li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span></code></pre></div>
<pre><code>## Le chargement a nécessité le package : Rcpp</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>database<span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">url</span>(<span class="st">&quot;https://figshare.com/ndownloader/files/30593259&quot;</span>))</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>gedepir<span class="sc">::</span><span class="fu">enrich</span>(results_NMF<span class="sc">$</span>T_matrix,<span class="at">pathways =</span> database,<span class="at">ICAbased =</span> <span class="cn">FALSE</span>, <span class="at">fdr =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL
## 
## [[3]]
## NULL
## 
## [[4]]
##                         pathway      padj       pval        ES
## 7 Kidney.Normal_cell.Neutrophil 0.3259838 0.04074798 0.9544087
## 
## [[5]]
## NULL
## 
## [[6]]
## NULL
## 
## [[7]]
## NULL
## 
## [[8]]
## NULL
## 
## [[9]]
##                                    pathway      padj       pval        ES
## 8 Brain.Normal_cell.Lake_et_al.Science.Ex8 0.1464789 0.01830986 0.9107322</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>database<span class="ot">=</span> <span class="fu">readRDS</span>(<span class="fu">url</span>(<span class="st">&quot;https://figshare.com/ndownloader/files/30593259&quot;</span>))</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>gedepir<span class="sc">::</span><span class="fu">enrich</span>(results_NMF<span class="sc">$</span>T_matrix,<span class="at">pathways =</span> database,<span class="at">ICAbased =</span> <span class="cn">FALSE</span>, <span class="at">fdr =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
## NULL
## 
## [[3]]
## NULL
## 
## [[4]]
##                         pathway      padj       pval        ES
## 7 Kidney.Normal_cell.Neutrophil 0.2987158 0.03733948 0.9544087
## 
## [[5]]
## NULL
## 
## [[6]]
## NULL
## 
## [[7]]
## NULL
## 
## [[8]]
## NULL
## 
## [[9]]
##                                    pathway      padj       pval        ES
## 8 Brain.Normal_cell.Lake_et_al.Science.Ex8 0.1353173 0.01691466 0.9107322</code></pre>
</div>
<div id="step-6-build-your-pipeline" class="section level2">
<h2>Step 6: Build your pipeline</h2>
<p><strong>gedepir</strong> simplifies the definition of a end-to-end analysis pipeline with a set of base functions that are connected through the <em>pipes</em> syntax <code>%&gt;%</code> used in <strong>magrittr</strong>, <strong>tidyr</strong> or <strong>dplyr</strong> packages.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>results_ICA<span class="ot">=</span> D <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_norm</span>(<span class="at">method =</span> <span class="st">&quot;RPM&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_trans</span>(<span class="at">method =</span> <span class="st">&quot;linear&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">run_deconv</span>(<span class="at">method =</span> <span class="st">&quot;ICA&quot;</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>gedepir<span class="sc">::</span><span class="fu">enrich</span>(results_ICA<span class="sc">$</span>T_matrix,database,<span class="at">ICAbased =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [[1]]
## NULL
## 
## [[2]]
##                                                   pathway       padj
## 500 Large_intestine.Normal_cell.DCLK1plus_progenitor_cell 0.04768075
##             pval       ES
## 500 8.637816e-05 0.963137
## 
## [[3]]
##                                               pathway       padj         pval
## 486                   Lung.Normal_cell.FOXN4plus_cell 0.02514119 0.0005216786
## 4   Chorionic_villus.Normal_cell.Villous_stromal_cell 0.04913960 0.0037673130
## 448        Embryo.Normal_cell.Primitive_endoderm_cell 0.04913960 0.0026603272
##            ES
## 486 0.7743916
## 4   0.9876944
## 448 0.7197438
## 
## [[4]]
##                                                      pathway       padj
## 15                    Undefined.Cancer_cell.Cancer_stem_cell 0.01681666
## 361                   Undefined.Cancer_cell.Cancer_stem_cell 0.01681666
## 151                    Breast.Normal_cell.Myoepithelial_cell 0.01725855
## 451      Embryonic_prefrontal_cortex.Normal_cell.Interneuron 0.01938336
## 159 Sclerocorneal_tissue.Normal_cell.Limbal_mesenchymal_cell 0.02263805
## 357                       Brain.Cancer_cell.Cancer_stem_cell 0.03167559
## 551                Embryo.Normal_cell.Morula_cell_Blastomere 0.03359531
## 190                      Breast.Cancer_cell.Cancer_stem_cell 0.03754822
## 389                   Undefined.Cancer_cell.Cancer_stem_cell 0.03754822
## 177  Peripheral_blood.Cancer_cell.Circulating_stem_like_cell 0.04158989
##             pval        ES
## 15  0.0003318951 0.9978099
## 361 0.0003318951 0.9978099
## 151 0.0014382122 0.9933920
## 451 0.0027677830 0.8848293
## 159 0.0035679533 0.9761107
## 357 0.0057117750 0.9782571
## 551 0.0061469684 0.7451756
## 190 0.0076889279 0.9742810
## 389 0.0077448283 0.9499071
## 177 0.0087399049 0.9832076
## 
## [[5]]
##                                                          pathway       padj
## 282                                Blood.Cancer_cell.Cancer_cell 0.01287073
## 289                           Ovary.Normal_cell.Endothelial_cell 0.01370959
## 463                            Ovary.Cancer_cell.Epithelial_cell 0.01370959
## 280 Hair_follicle.Normal_cell.Epithelial_hair_follicle_stem_cell 0.01460157
## 46                            Thymus.Normal_cell.Epithelial_cell 0.01638093
## 483                              Lung.Normal_cell.Secretory_cell 0.01654460
## 240                            Liver.Cancer_cell.Liver_stem_cell 0.02298157
## 326                      Bone_marrow.Normal_cell.Langerhans_cell 0.02437086
## 276                        Undefined.Normal_cell.Epithelial_cell 0.02494178
## 281                                Blood.Cancer_cell.Cancer_cell 0.02494178
##             pval        ES
## 282 9.681479e-05 0.9915625
## 289 4.574984e-04 0.9684360
## 463 4.574984e-04 0.9684360
## 280 6.487188e-04 0.9909539
## 46  1.259975e-03 0.9811640
## 483 1.408689e-03 0.7664802
## 240 2.414730e-03 0.9661776
## 326 2.817185e-03 0.9650088
## 276 3.017241e-03 0.9903225
## 281 3.027354e-03 0.9804395</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
